ckanext/unaids/tests/test_plugin.py

```diff
@@ -42,52 +42,85 @@ class TestResourceLastModified(object):
     whenever a resource is created or edited.
     '''
 
-    URL_TYPE_FILE_UPLOAD = 'upload'
-    URL_TYPE_LINK = ''
-
-    def test_resource(self, url_type):
-        resource_dict = {'url_type': url_type}
-        if url_type == self.URL_TYPE_FILE_UPLOAD:
-            resource_dict.update({
-                'url': 'file.csv',
-                'lfs_prefix': 'prefix123',
-                'sha256': 'sha256',
-                'size': 100,
-            })
-        elif url_type == self.URL_TYPE_LINK:
-            resource_dict.update({
-                'url': 'http://example.com'
-            })
-        return resource_dict
-
-    def test_null_to_file(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_file_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_file_to_url(self):
-        current = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
-
-    def test_null_to_url(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date(resource=resource)
-        assert 'last_modified' in resource
-
-    def test_url_to_null(self):
-        resource = self.test_resource(self.URL_TYPE_LINK)
-        _update_resource_last_modified_date({}, resource)
-        assert 'last_modified' not in resource
-
-    def test_url_to_file(self):
-        current = self.test_resource(self.URL_TYPE_LINK)
-        resource = self.test_resource(self.URL_TYPE_FILE_UPLOAD)
-        _update_resource_last_modified_date(resource, current)
-        assert 'last_modified' in resource
+    @pytest.fixture
+    def resource_with_file(autouse=True):
+        return {
+            'url_type': 'upload',
+            'url': 'file.csv',
+            'lfs_prefix': 'prefix123',
+            'sha256': 'sha256',
+            'size': 100,
+        }
+
+    @pytest.fixture
+    def resource_with_link(autouse=True):
+        return {
+            'url_type': '',
+            'url': 'http://example.com'
+        }
+
+    def test_null_to_file(self, resource_with_file):
```

Comment:
Could you update all test method to include the desired outcome of the test?
This will document the test and make test reports (and failure reports!) more descriptive. The latter is especially important if you get an error report from CI (github actions, jenkins etc.)

Generally: `test_[explain init situation]_should_[provide expected result]`
here: `test_null_to_file_should_update_last_modified_datetime`

This should be applied to all test cases in the class.